### DISCUSSION TITLE
[Feature] Batch operations for albums

### DISCUSSION THREAD

**Discussion by chrneu** [immich-app/immich#8543] [Feature Request]
### The feature

Right now a user is able to select multiple photos and do batch operations like change time, change location, archive, share.

It would be nice to do this on album level, maybe even on a nested album level when available.

Especially for the location topic this would save a lot of time, while cameras often don't store this information so you have to maintain this after you indexed an external library.

Also this functionality should not be possible for shared albums. I just tried to delete a shared album from a created test user which gave me only the error message "Error deleting album". I would have expected something like "You don't have the permission to delete an album which was shared with you." or not even giving the option in the menu. Instead of delete, better give the option to "Leave shared album".

### Platform

- [x] Server
- [X] Web
- [ ] Mobile

**Comment by Thinkscape** [comment]
Yes please!!!
I've got ~100 albums, and want to share all of them with my family.
There is no reasonable way to do that right now, as it currently requires manually navigating to each and manually selecting each person to share with.

**Comment by agster27** [comment]
I have over a 100 albums created by mistake and I would love this feature to bulk select and delete them.

**Reply by vinnividivicci** [reply]
👆🏻this

**Reply by dmitrychepel** [reply]
Function "add partner" is not a replacement of shared albums now.
I think that best way to merge this functionality into one.

So the best option for Funciton "add partner" I mean reimplement it throught add sharing to every albums with partner.
And certanly add bulk manage access for albums.

This is also simplify using mobile app in two roles 
 - mobile backup
 - view server photo

**Comment by troyliu0105** [comment]
I find a simple way to delete albums in batch using api via `jq` and `curl`.

```bash
curl -L "https://immich.com/api/albums" \
         -H 'Accept: application/json' \
         -H 'x-api-key: abcde' | jq '.[] | select(.albumName | test("^[a-zA-Z0-9]{4}$")) | .id' \
         | xargs -I {} \
         curl -L -X DELETE "https://immich.com/api/albums/{}" \
         -H 'x-api-key: abcde'
```

you can adjust the `jq` selector to find the albums you wanna delete.

**Comment by thisisgonnatakeawhile** [comment]
+1. This could come in handy when attempting to delete multiple non-shared albums.
Having a prompt to delete all the media within the selected albums could also facilitate usability.
